### PR TITLE
[UI] Update Challenge UI

### PR DESCRIPTION
**Pull request by brain-frog**
## What are the changes?

Making the challenge UI a little nicer to look at

## Why am I doing these changes?

I didn't get a chance to review the challenge UI before it went live

## What did change?

all challenge localization
how text shows up for descriptions of challenges
let types for the type challenge be a sprite

### Screenshots/Videos

https://github.com/pagefaultgames/pokerogue/assets/15065660/13c90449-bfa4-428a-a753-a52844352c66

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Dakurei** [comment]
I've tested it locally, and it works very well for me 👍

As the previous translation was retrieved despite the changes made, it's already okay for French, so no retouching necessary 👌

https://github.com/pagefaultgames/pokerogue/assets/8146474/0b00e13e-6146-43c6-93a7-31c32aa94b32


👉 **If anyone involved in the other translations would like to test, we'd be delighted to receive your help and feedback.**

File: src/locales/de/challenges.ts [hunk 1]
```diff
@@ -1,67 +1,25 @@
...
-  "singleType.value.15": "Eis",
-  "singleType.desc.15": "Du kannst nur Pokémon des Typs Eis verwenden.",
-  "singleType.value.16": "Drache",
-  "singleType.desc.16": "Du kannst nur Pokémon des Typs Drache verwenden.",
-  "singleType.value.17": "Unlicht",
-  "singleType.desc.17": "Du kannst nur Pokémon des Typs Unlicht verwenden.",
-  "singleType.value.18": "Fee",
-  "singleType.desc.18": "Du kannst nur Pokémon des Typs Fee verwenden."
-
+  "illegalEvolution": "{{pokemon}} changed into an ineligble pokémon\nfor this challenge!",
```
Comment (CodeTappert) [inline src/locales/de/challenges.ts:5]:
```suggestion
  "illegalEvolution": "{{pokemon}} hat sich in ein Pokémon verwandelt, dass für diese Herausforderung nicht zulässig ist!",
```

File: src/locales/de/challenges.ts [hunk 2]
```diff
@@ -1,67 +1,25 @@
...
+  "singleGeneration": {
+    "name": "Mono-Generation",
+    "desc": "Du kannst nur Pokémon aus der {{gen}} Generation verwenden.",
+    "desc_default": "Du kannst nur Pokémon gewählten Generation verwenden.",
+    "gen_1": "ersten",
+    "gen_2": "zweiten",
```
Comment (CodeTappert) [inline src/locales/de/challenges.ts:11]:
```suggestion
    "gen_2": "zweiten",
    "gen_3": "dritten",
```

Comment (CodeTappert) [reply src/locales/de/challenges.ts:11]: You are missing the 3rd gen in german :D

**Comment by CodeTappert** [comment]
So i normally just wanted to test the german translation i found a bug. But the bug isnt part of this so i can approve this current PR. The translation is good
https://github.com/pagefaultgames/pokerogue/assets/38758606/85b001ed-cd5f-41ac-a839-487849f3c85e

**Review by CodeTappert — APPROVED** [review]
Even though my approve doesnt change anything. I would say this PR is ready to merge

**Review by XGSleepWalker — APPROVED** [review]

**Pull request merged**